Sources/Security.php

```diff
@@ -1474,4 +1448,33 @@ function corsPolicyHeader($set_header = true)
 	}
 }
 
+/**
+ * Helper function to figure out a urls base domain.
+ *
+ * @since 2.1
+ */
+function FindCorsBaseUrl($url, $sub_domain = false)
+{
+	global $modSettings;
+
+	$base_domain = parse_url($url, PHP_URL_HOST);
+
+	// If we find www, pop it out.
+	if ('www' === array_slice(explode('.', $base_domain), 0, 1))
+		$base_domain = implode('.', array_slice(explode('.', $base_domain, 1)), 1);
+
+	// Are we saying this is a sub domain and to remove another domain level?
+	if ($sub_domain)
+		$base_domain = implode('.', array_slice(explode('.', parse_url($domain, PHP_URL_HOST)), 1));
+
+	/*	Note, we do have a TLD regex for the autolinker, however it can not reliably be used here due to
+		Country Code Second Level Domains (example: co.uk) as they are not TLDs.  As these are also
```

Comment:
Could the list be used here as well?
https://publicsuffix.org/
https://github.com/SimpleMachines/SMF2.1/pull/6764

Reply:
I saw it.  I believe that this could work but the benefit doesn't seem to be wroth all the work to determine the limited number of hosts.  I guess we could upon saving of the cors settings, we could rebuild a hidden list of "true" domains if it seems its better for us to figure out the real root domain.